source/nn_memory_bank.py

```diff
@@ -9,16 +9,15 @@ def __init__(self, size: int = 2 ** 16):
     def forward(self,
                 output: torch.Tensor,
                 labels: torch.Tensor = None,
-                update: bool = False,
-                normalize: bool = True):
-
-        if normalize:
-            output = torch.nn.functional.normalize(output, dim=1)
+                update: bool = False):
 
         output, bank = super(NNmemoryBankModule, self).forward(output, labels, update)
         bank = bank.to(output.device).t()
 
-        similarity_matrix = torch.einsum("nd,md->nm", output, bank)
+        output_normed = torch.nn.functional.normalize(output, dim=1)
+        bank_normed = torch.nn.functional.normalize(bank, dim=1)
```

Comment:
Is there a reason for moving the normalisation after the memory bank instead of before it? The memory bank is already initialised normally, thus this should not be the reason. This implementation of normalising the bank every time is more compute-intensive.

Reply:
I just didn't want to enforce the normalization of the returned values.